File: introduction/removing_duplicates.cpp
#include<iostream>
#include<vector>
#include<set>
using namespace std;


vector<int> removeDuplicates(vector<int> input){

    // Write your code here
  set<int> s;
  vector<int> result;
  for(int i=0;i<input.size();i++){
	if(s.find(input[i])==s.end()){
		s.insert(input[i]);
		result.push_back(input[i]);		
		}
	}
return result;
}


//this is optional lines of code for coding ninjas 
int main(){
	int n;
	cin>>n;
	vector<int> input(n);
	vector<int> result;
	for(int i=0;i<input.size();i++)
		cin>>input[i];
	result=removingDuplicates(input);
	for(int i=0;i<result.size();i++)
		cout<<result[i];
	return 0;
	
}

File: introduction/removing_duplicates.java
import java.util.*;

/*
	- You are given a sorted integer array that can contain duplicate entries, i.e., same value might appear more than once.

	- Your task is to create a new sorted array that will not contain any duplicates, i.e., the repeating elements should appear only once.

*/

public class solution 
{
	public static int[] removeDuplicates(int arr[])
	{
		//	write your code here.
      HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < arr.length; i++) {
            set.add(arr[i]);
        }

        // Integer[] arr1 = new Integer[set.size()];

        // arr1 = set.toArray(arr1);
        int[] arr1 = new int[set.size()];
        int j=0;
        for (Integer i : set) {
            arr1[j] = i.intValue();
            j++;
        }
        

        Arrays.sort(arr1);
		return arr1;
	}
}
/*this is a complete code for the problem of coding ninjas*/
